Remove candidate passers w/o feasible lever

Lolligerhans opened the pull request:
Objections from anyone to code style would best be raised now. Specifically regarding

1. the way I nest operations in lengthy instructions and
2. the use of `const` to help readers and compiler to better understand the computations.

Commit message will not show the fancy ASCII if viewed without monospace font. I like including the pictures though - nobody should be bothered reading text or even diff. The text will remain readable. Wrapped at 50/72.

Testing on takes 4 and 5 is ongoing but not promising.

I add a comment in `pawns.cpp` to remind readers that identification of candidate passers continues in `evaluate.cpp`.

I use the phrasing "candidate passer" with a meaning _distinct_ from the one used in recent "Retire candidate passed pawns" commit, which abandoned spacial handling of a different form of "candidate passer".

You do not need to edit this commit. Comment here - I will edit the branch as required.

Lolligerhans commented:
Updated bench.

vondele commented:
nice idea... for the formatting, try to blend in the code as much as possible, i.e. use same indentation style (which tends to align near = signs). That's a little more difficult with the const decl, which presumably for variables with such a local scope doesn't bring much neither for the compiler nor for the reader.

Alayan-stk-2 commented:
Excellent commit message.

snicolet commented:
@Lolligerhans 

Nice idea ! Concerning the formatting, I suggest something like the following version where I used only two bitboards, declaring them at the top of the function as usual --  didn't verified that the bench is the same, however, please double-check -- and put a singular in the comment:

``` C++

Bitboard b, bb, squaresToQueen, unsafeSquares, candidatePassers, leverable;

...

candidatePassers = b & shift<Down>(pos.pieces(Them, PAWN));

if (candidatePassers)
{
    // Can we lever the blocker of a candidate passer?
    leverable =   shift<Up>(pos.pieces(Us, PAWN))
               &  ~pos.pieces(Them)
               & (~attackedBy2[Them] | attackedBy[Us][ALL_PIECES])
               & (~(attackedBy[Them][KNIGHT] | attackedBy[Them][BISHOP])
                 | (attackedBy[Us  ][KNIGHT] | attackedBy[Us  ][BISHOP]));

    // Remove candidate otherwise
    b &= ~(candidatePassers & ~(shift<WEST>(leverable) | shift<EAST>(leverable));
}
```

Lolligerhans commented:
- Align at equal sign as suggested by vdv and sn.
- Squeezed removal `&=` into 1 expression. Decided to de morgan it (line exceeded 80c).
- Fixed typo in comment using plural.
- Removed all `const` and moved declarations to start.
- Double-checked: executable remains identical.

vondele commented:
LGTM, will merge in the next round. 

The commit message has one more typo (`s/althogh/although/g`) and maybe you can use the same format (including link location) for the SPRT results as most other commit messages.

Lolligerhans commented:
- Fixed altho(u)gh and some bracing in commit message text.
- Adjusted test report formatting. Found multiple variants, including ones omitting links. Copied what seems the most reasonable.

adentong commented:
Really nice commit message! Good stuff.

vondele commented:
Thanks!
